lib/duplicate_deleter.rb

```diff
@@ -77,7 +77,8 @@ def call(ids, id_type: 'content_id')
         next
       end
 
-      Indexer::DeleteWorker.new.perform(index_names.first, type_to_delete, ids.first)
+      id_to_delete = results[:results].detect { |a| a[:elasticsearch_type] == type_to_delete }[:_id]
```

Comment:
Line 52 should guard against that case, since it checks whether any of the results have `type_to_delete` type.

I've added an extra nil check anyway, since there's a lot of logic in this class and an error message might help someone track down the problem.